### PR TITLE
Auth final

### PR DESCRIPTION
**Pull request by P3BBL35**
Final part to the authentication library! Next up is either charts or maps...might use a coin flip to decide which one as both are equally interesting.

### CODE REVIEW

**Comment by daperlmut** [comment]
> Final part to the authentication library! Next up is either charts or maps...might use a coin flip to decide which one as both are equally interesting.

Did you know that if you type "flip a coin" into Google it will do it for you.  It even has an animation :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -143,7 +145,7 @@ private boolean getToChange(HttpServletRequest request) {
   }
 
   /**
-   * @return the display name of the currently logged-in user.
+   * @return the display name of the currently logged-in user.
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:148]: [Nit] Remove extra whitespace

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -27,18 +27,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     UserService userService = UserServiceFactory.getUserService();
 
     List<String> list = new ArrayList<>();
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:29]: We should probably switch this to use a more informative name than "list".  Maybe something like "jsonElements"?  I'm not in love with that name so if you have a better idea that would be fine as well.

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -27,18 +27,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:30]: [Optional] The extra newline here seems unnecessary to me

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 4]
```diff
@@ -27,18 +27,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    list.add(String.valueOf(userService.isUserLoggedIn()));
+
     if (userService.isUserLoggedIn()) {
       String userEmail = userService.getCurrentUser().getEmail();
       String urlToRedirectToAfterUserLogsOut = "/comments.html";
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:35]: [Optional] It would be fine to hard pass this directly as a constant into the function below rather than creating a variable that is used only once.

Same for 'urlToRedirectToAfterUserLogsIn' below.

File: portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+@WebServlet("/nickname")
+public class NicknameServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    String name = request.getParameter("nickname");
+
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java:28]: [Optional] Extra newline seem extraneous

File: portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      final String userId = userService.getCurrentUser().getUserId();
+
+      Entity entity = new Entity("User", userId);
+      entity.setProperty("id", userId);
+      entity.setProperty("name", name);
+
+      datastore.put(entity);
+    }
+
+    response.sendRedirect("/comments.html");
```
Comment (daperlmut) [inline portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java:41]: [Optional] Extra newline seem extraneous

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -129,29 +127,42 @@ function getOrder() {
  */
 function isLoggedIn() {
 
-  fetch('/login').then(response => response.json()).then((status) => {
-    let loginStatus = status[0];
+  fetch('/login').then(response => response.json()).then((stat) => {
```
Comment (daperlmut) [inline portfolio/src/main/webapp/script.js:130]: [Nit] Why did you change 'status' to 'stat'?  I think 'status' is more readable?

Comment (P3BBL35) [reply portfolio/src/main/webapp/script.js:130]: status was marked as a keyword by VIM, so I changed it to avoid that. However, it seemed to be working fine before, so I will switch back and test again.

Comment (daperlmut) [reply portfolio/src/main/webapp/script.js:130]: It does appear to be reserved
https://www.w3schools.com/JS/js_reserved.asp

In that case, I would rather use a longer name instead of a shorter one.  Also, it appears that this variable stores all the JSON info so maybe call it something like 'jsonUserInfo'
